packages/components/src/tools-panel/tools-panel/hook.ts

```diff
@@ -7,21 +7,32 @@ import { useEffect, useMemo, useRef, useState } from '@wordpress/element';
  * Internal dependencies
  */
 import * as styles from '../styles';
-import { useContextSystem } from '../../ui/context';
+import { useContextSystem, WordPressComponentProps } from '../../ui/context';
 import { useCx } from '../../utils/hooks/use-cx';
-
-const generateMenuItems = ( { panelItems, reset } ) => {
-	const menuItems = { default: {}, optional: {} };
-
-	panelItems.forEach( ( { hasValue, isShownByDefault, label } ) => {
+import type {
+	ToolsPanelItem,
+	ToolsPanelMenuItemKey,
+	ToolsPanelMenuItems,
+	ToolsPanelMenuItemsConfig,
+	ToolsPanelProps,
+} from '../types';
+
+const generateMenuItems = ( config: ToolsPanelMenuItemsConfig ) => {
+	const { panelItems, reset } = config;
+	const menuItems: ToolsPanelMenuItems = { default: {}, optional: {} };
+
+	panelItems.forEach( ( panelItem: ToolsPanelItem ) => {
+		const { hasValue, isShownByDefault, label } = panelItem;
```

Comment:
The changes to these lines are not necessary, since TypeScript knows that `panelItems` is of type `ToolsPanelItem[]`. The previous version of the code should still work fine:

```suggestion
	panelItems.forEach( ( { hasValue, isShownByDefault, label } ) => {
```